projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.native.js

```diff
@@ -63,20 +73,72 @@ export default function Player( { isSelected, attributes } ) {
 		poster,
 	} );
 
-	const { preview, isRequestingEmbedPreview } = usePreview( videoPressUrl );
-	const html = addTokenIntoIframeSource( preview?.html, token );
+	// Helper to invalidate the preview cache.
+	const invalidateResolution = useDispatch( coreStore ).invalidateResolution;
+	const invalidatePreview = useCallback( () => {
+		invalidateResolution( 'getEmbedPreview', [ videoPressUrl ] );
+	}, [ videoPressUrl, invalidateResolution ] );
+
+	const { preview = {}, isRequestingEmbedPreview } = usePreview( videoPressUrl );
+	const html = addTokenIntoIframeSource( preview.html, token );
+
+	const cancelPreviewCheckInterval = useCallback( () => {
+		clearInterval( previewCheckInterval.current );
+	}, [ previewCheckInterval ] );
+
+	// Fetch the preview until it's ready
+	useEffect( () => {
+		// Wait for the client player to load
+		if ( ! isPlayerLoaded ) {
+			return;
+		}
+		previewCheckInterval.current = setInterval( () => {
+			// if the preview is ready or we reached the max attempts, clear the interval
+			if ( isPreviewReady( preview ) || previewCheckAttempts > VIDEO_PREVIEW_ATTEMPTS_LIMIT ) {
+				return cancelPreviewCheckInterval();
+			}
+
+			// Avoid lapping any prior request
+			if ( isRequestingEmbedPreview ) {
+				return;
+			}
+
+			// Invalidate the cache and wait for the next interval since the preview is not ready yet
+			setPreviewCheckAttempts( previewCheckAttempts + 1 );
+			invalidatePreview();
+		}, 1000 );
+
+		return cancelPreviewCheckInterval;
+	}, [ preview, isPlayerLoaded ] );
+
+	const onSandboxMessage = useCallback( message => {
+		if ( message.event === 'videopress_loading_state' && message.state === 'loaded' ) {
+			setIsPlayerLoaded( true );
+		}
+	}, [] );
 
-	// Set up style for when the player is loading.
 	const loadingStyle = {};
-	if ( ! html || isRequestingEmbedPreview ) {
+	if ( ! isPreviewReady( preview ) ) {
 		loadingStyle.height = 250;
 	}
 
+	const renderEmbed = () => {
+		if ( html ) {
+			return (
+				<SandBox
+					html={ html }
+					onWindowEvents={ { message: onSandboxMessage } }
+					viewportProps="user-scalable=0"
+				/>
+			);
+		}
+		return <Text>{ __( 'Loading…', 'jetpack-videopress-pkg' ) }</Text>;
+	};
+
 	return (
 		<View style={ [ style[ 'videopress-player' ], loadingStyle ] }>
 			{ ! isSelected && <View style={ style[ 'videopress-player__overlay' ] } /> }
-			{ ! isRequestingEmbedPreview && <SandBox html={ html } viewportProps="user-scalable=0" /> }
-			{ ! html && <Text>{ __( 'Loading…', 'jetpack-videopress-pkg' ) }</Text> }
```

Comment:
The `! isRequestingEmbedPreview` vs `! html` checks here are logically off. These two components should toggle between one state. I considered showing the loading component until the preview is ready but there would be an odd flicker between the VP processing screen and the video. 

This is something we could consider if we don't want to see the cropped video. After I fix https://github.com/wordpress-mobile/gutenberg-mobile/issues/5654 I can get a screen recording of this option.